### ISSUE TITLE
Upgrade 8.15.0 -> 8.16.0

### ISSUE THREAD

**Issue by cpoetter** [getsentry/sentry#5336]
I just upgraded from Sentry **8.15.0** to **8.16.0**. I destroyed my *Sentry Server*, *Sentry Cron* and *Sentry Worker* Docker container. Started only my upgraded *Sentry Server* container and ran `sentry upgrade` in it. After that I started the new *Sentry Cron* and *Sentry Worker* container. 

Once I logged in to sentry I saw these error messages:

<img width="1185" alt="warnings" src="https://cloud.githubusercontent.com/assets/9142522/25674672/00ffa352-303c-11e7-8a96-5477addd7324.png">

```
ProgrammingError('column sentry_useroption.organization_id does not exist\nLINE 1: ...ion"."user_id", "sentry_useroption"."project_id", "sentry_us...\n ^\n',) SQL: SELECT "sentry_useroption"."id", "sentry_useroption"."user_id", "sentry_useroption"."project_id", "sentry_useroption"."organization_id", "sentry_useroption"."key", "sentry_useroption"."value" FROM "sentry_useroption" WHERE ("sentry_useroption"."project_id" IS NULL AND "sentry_useroption"."user_id" = %s )
```
Is that *expected* behavior after an upgrade? It's the first time I make an update ...

**Comment by mattrobenolt** [comment]
This isn't expected. It looks like you didn't run the migrations needed for 8.16.0 because the schema is incorrect.

**Comment by cpoetter** [comment]
Hm I ran the migration **three** times just to be sure it worked :D 

Should I have maybe started the container like
```
docker run -it --rm -e SENTRY_SECRET_KEY='' --link postgres:postgres --link redis:redis sentry upgrade
```
instead of 
```
docker run -e SENTRY_SECRET_KEY='' --link postgres:postgres --link redis:redis sentry
```
and then logging in and run the `sentry upgrade` then?

Thanks!

**Comment by mattrobenolt** [comment]
No, just all I can say is that the `sentry` container you're running the migration with has to be old. I'd try to be explicit with `sentry:8.16` tag instead.

**Comment by cpoetter** [comment]
I'm kinda sure it's version 8.16.0. At least that's what it says once I logged in, see screenshot.

<img width="1014" alt="version" src="https://cloud.githubusercontent.com/assets/9142522/25675653/8d301cfa-303f-11e7-81c8-5abd5a198955.png">

Can I have an old image and it anyway shows the new version number? No, correct?

Were these errors maybe reported in the small time between starting the container and manually running the upgrade command? That's why I had the question if I need to start the container from the new image the first time with the `update` command at it's end to prevent this small time of mismatching image and database.

**Comment by mattrobenolt** [comment]
> Were these errors maybe reported in the small time between starting the container and manually running the upgrade command? 

Oh, then if this is the case, yeah, that'd be expected. The new instances require the schema migrations.

Ideally, you can get away with running `upgrade` from a new version before bringing up new versions of everything else. That's not always true in practice, but there will always be some partial interruption between upgrading.

If the errors aren't continuing, then you're good.

**Comment by cpoetter** [comment]
Okay, thanks @mattrobenolt for your super fast help! So when the next upgrade happens I make it the same way like now? Start only *Sentry Server*, login and run `sentry upgrade` inside? After that start the other containers?

Is the `docker run .... sentry upgrade` command only for the **really first install**?

**Comment by mattrobenolt** [comment]
> Is the docker run .... sentry upgrade command only for the really first install?

I'd just do this all the time.

**Comment by cpoetter** [comment]
Awesome, thanks!

**Comment by cpoetter** [comment]
@mattrobenolt Just wanted to let you know that 
```
docker run -e SENTRY_SECRET_KEY='...' --link postgres:postgres --link redis:redis sentry upgrade
```
does not work for **always**. After the migration finishes the instance shuts down. And because I set `auto restart = true` my instance was restarting and migration all the time. So this really just works once for an upgrade.

Maybe it should be mentioned in the Docker Hub file that the `docker run ... sentry upgrade` command should be run after an upgrade, so that people don't see the errors I saw (between starting the instance and running the migration manually).

In any case: my problem is solved! Thanks for your help!
